[Merged by Bors] - chore(Tactic/Linters): remove `public section`

grunweg opened the pull request:
Mostly as a learning exercise for me to understand the module system better.

---

- [x] depends on: #31823 (to make visibility changes just for tests a conscious choice)

<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

When merging, all the commits will be squashed into a single commit
listing all co-authors.

Co-authors in the squash commit are gathered from two sources:

First, all authors of commits to this PR branch are included. Thus,
one way to add co-authors is to include at least one commit authored by
each co-author among the commits in the pull request. If necessary, you
may create empty commits to indicate co-authorship, using commands like so:

git commit --author="Author Name <author@email.com>" --allow-empty -m "add Author Name as coauthor"

Second, co-authors can also be listed in lines at the very bottom of
the commit message (that is, directly before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines
at the bottom of the commit message (before the `---`, and also before
any "Co-authored-by" lines) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

grunweg committed:
chore(DeprecatedSyntaxLinter): privatise

grunweg committed:
chore: try DeprecatedModule also

grunweg committed:
A few more

grunweg committed:
chore(TextBased): two unit tests require importing further definitions;
should MathlibTest also allow the module system instead?

grunweg committed:
A few more: do I need to make macros public?

grunweg committed:
Fix style lint

grunweg committed:
style: use meta section instead

grunweg committed:
Fix tests by making more information public

grunweg committed:
A few more

grunweg committed:
And the directoryDependency linter

grunweg committed:
HaveLet linter

grunweg committed:
Fix style lint

grunweg commented:
(Just in case, I would be a bit surprised to see any effect.)

JovanGerb commented on Mathlib/Tactic/Linter/CommandStart.lean:
```diff
@@ -193,6 +193,7 @@ def parallelScanAux (as : Array FormatError) (L M : String) : Array FormatError
       pushFormatError as (mkFormatError ls ms "Oh no! (Unreachable?)")
 
 @[inherit_doc parallelScanAux]
+public -- for use in unit tests only
```
I think you should be able to `meta import all` in the test file. Or does that not work? (Similar below)

grunweg commented on Mathlib/Tactic/Linter/DeprecatedSyntaxLinter.lean:
```diff
@@ -235,3 +235,5 @@ def deprecatedSyntaxLinter : Linter where run stx := do
 
 end Mathlib.Linter.Style
+
+end
```
I'm not sure what the prevailing style is. We don't enforce a closing section (because of pre-existing exceptions), but that's not the same.

grunweg replied:
Sure! I guess that was done more for convenience than anything (and was going by analogy with `noncomputable section`, not `expose section`...

grunweg committed:
chore: expose less by making tests use import_all

grunweg commented on Mathlib/Tactic/Linter/DirectoryDependency.lean:
```diff
@@ -96,7 +98,8 @@ prefixes `n₁'` of `n₁` and `n₂'` of `n₂` such that `n₁' R n₂'`.
 turn. This can probably be optimized.
 -/
-@[expose] def NamePrefixRel := NameMap NameSet
+--@[expose]
```
```suggestion
```

grunweg commented on Mathlib/Tactic/Linter/DirectoryDependency.lean:
```diff
@@ -96,7 +98,8 @@ prefixes `n₁'` of `n₁` and `n₂'` of `n₂` such that `n₁' R n₂'`.
 -/
-@[expose] def NamePrefixRel := NameMap NameSet
+--@[expose]
+def NamePrefixRel := NameMap NameSet
```
I'm removing this because this doesn't seem needed. I'm happy to add it back.

grunweg reviewed (commented):
I've audited all public's for some that should be private: ~~only the flexible linter remains.~~ update: now done also

grunweg committed:
Fix flexible linter also

grunweg committed:
change: make modulesNotUpperCamelCase and modulesOSForbidden public again

Their implementation might be subject to changes, but the signature feels
stable enough.

Vierkantor commented on Mathlib/Tactic/Linter/Header.lean:
```diff
@@ -76,8 +76,7 @@ It returns the array of all `import` identifiers in `s`. -/
 --   which we use to log more precise warnings.
-partial
-def getImportIds (s : Syntax) : Array Syntax :=
+public partial def getImportIds (s : Syntax) : Array Syntax :=
```
(Could you add a comment explaining) why this declaration is public? It looks like this is something internal to the linter.

Vierkantor reviewed (approved):
Looks good to me otherwise, nice work!

grunweg committed:
Update Mathlib/Tactic/Linter/DirectoryDependency.lean

grunweg committed:
Review comments

grunweg commented:
-awaiting-author

jcommelin reviewed (commented):
Thanks :tada:

bors merge
